File: Factory Pattern/Cpp/PizzaStore/ChicagoPizzaStore.cpp
#include "../Pizza/Pizza.cpp"
#include "../Pizza/ChicagoStylePizza/ChicagoStyleCheesePizza.cpp"
#include "../Pizza/ChicagoStylePizza/ChicagoStyleClamPizza.cpp"
#include "../Pizza/ChicagoStylePizza/ChicagoStyleMeatPizza.cpp"
#include "../Pizza/ChicagoStylePizza/ChicagoStyleVeggiePizza.cpp"

#include "PizzaStore.cpp"

class ChicagoPizzaStore : public PizzaStore {
    Pizza* createPizza(string type) override {
        if (type == "cheese") return new ChicagoStyleCheesePizza();
        else if (type == "clam") return new ChicagoStyleClamPizza();
        else if (type == "meat") return new ChicagoStyleMeatPizza();
        else if (type == "veggie") return new ChicagoStyleVeggiePizza();
        else return nullptr;
    }
};

File: Patterns/Factory Method Pattern/CSharp/PizzaStore/ChicagoPizzaStore.cs
using Pizza;

using ChicagoStylePizza = Pizza.ChicagoStylePizza;

namespace PizzaStore
{
    public class ChicagoPizzaStore : PizzaStore 
    {
        protected override Pizza.Pizza? CreatePizza(string type)
        {
            switch (type) 
            {
                case "cheese" : return new ChicagoStylePizza.ChicagoStyleCheesePizza();
                case "clam" : return new ChicagoStylePizza.ChicagoStyleClamPizza();
                case "meat" : return new ChicagoStylePizza.ChicagoStyleMeatPizza();
                case "veggie" : return new ChicagoStylePizza.ChicagoStyleVeggiePizza();
            }

            return null;
        }
    }
}